pkg/riff/commands/build_status.go

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package commands
+
+import (
+	"fmt"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	"github.com/projectriff/riff/pkg/cli"
+	"strings"
+)
+
+func buildConditionStatusInfo(cond *duckv1alpha1.Condition) []map[string]string {
+	info := make([]map[string]string, 5)
+	info[0] = map[string]string{"Type": fmt.Sprintf("%s", cond.Type)}
+	info[1] = map[string]string{"Status": fmt.Sprintf("%s", cond.Status)}
+	info[2] = map[string]string{"Reason": cond.Reason}
+	info[3] = map[string]string{"Severity": fmt.Sprintf("%s", cond.Severity)}
+	info[4] = map[string]string{"Message": cond.Message}
+	return info
+}
+
+func printStatusMessage(c *cli.Config, resource string, info []map[string]string) {
+	c.Printf("%s:\n", strings.ToUpper(resource))
+	for i := 0; i < len(info); i++ {
+		key := firstKey(info[i])
+		c.Printf(" %s: %s\n", key, info[i][key])
+	}
+	c.Printf("\n")
+}
+
+func firstKey(m map[string]string) string {
```

Comment:
key iteration order for maps is not deterministic. Why use a map if you're not using it as a map?

Reply:
there is only one entry in the map - is there a better go type to use, couldn't find a tuple or pair to use, rewriting this with a struct